src/Utils.cpp

```diff
@@ -132,29 +132,50 @@ bool MCMUtils::GetPropertyValue(const char * formIdentifier, const char * proper
 	}
 
 	VirtualMachine* vm = (*G::gameVM)->m_virtualMachine;
-	VMValue targetScript;
-	PackValue(&targetScript, &targetForm, vm);
-	if (!targetScript.IsIdentifier()) {
+
+	IObjectHandlePolicy * hp = (*g_gameVM)->m_virtualMachine->GetHandlePolicy();
+	UInt64 hdl = hp->Create(targetForm->kTypeID, targetForm);
+
+	VirtualMachine::IdentifierItem * attachedScripts = (*g_gameVM)->m_virtualMachine->m_attachedScripts.Find(&hdl);
+
+	if (attachedScripts->count == 0)
+	{
 		_WARNING("Warning: Cannot retrieve a property value %s from a form with no scripts attached. (%s)", propertyName, formIdentifier);
-		return false;
+		return false;
+	}
+
+	VMIdentifier* targetScriptIdentifier = nullptr;
+
+	if (attachedScripts->count == 1)
+	{
+		targetScriptIdentifier = attachedScripts->GetScriptObject(attachedScripts->identifier.one);
+	}
+	else
+	{
+		for (UInt32 i = 0; i < attachedScripts->count; ++i)
+		{
+			targetScriptIdentifier = attachedScripts->GetScriptObject(attachedScripts->identifier.many[i]);
+			if (_stricmp(targetScriptIdentifier->m_typeInfo->m_typeName.c_str(), scriptName) == 0)

```

Comment:
What happens if `scriptName` is `nullptr`?

i.e. form has more than one script attached and no script name is specified